ee/server/service/digicert/digicert.go

```diff
@@ -73,6 +73,152 @@ func VerifyProfileID(ctx context.Context, logger kitlog.Logger, config fleet.Dig
+			"seat_id": config.CertificateSeatID,
+		},
+		"delivery_format": "x509",
+		"attributes": map[string]interface{}{
+			"subject": map[string]string{
+				"common_name": config.CertificateCommonName,
+			},
+		},
+		"csr": csr,
+	}
+
+	bodyBytes, err := json.Marshal(reqBody)
+	if err != nil {
+		return nil, "", ctxerr.Wrap(ctx, err, "marshaling request body")
+	}
+
+	config.URL = strings.TrimRight(config.URL, "/")
+	req, err := http.NewRequest("POST", config.URL+"/mpki/api/v1/certificate", strings.NewReader(string(bodyBytes)))
+	if err != nil {
+		return nil, "", ctxerr.Wrap(ctx, err, "creating DigiCert POST request")
+	}
+
+	req.Header.Set("X-API-key", config.APIToken)
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Accept", "application/json")
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, "", ctxerr.Wrap(ctx, err, "sending DigiCert POST request")
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusCreated {
+		// Try to see if errors are present in body
+		type errorResponse struct {
+			Errors []struct {
+				Code    string `json:"code"`
+				Message string `json:"message"`
+			} `json:"errors"`
+		}
+		var errResp errorResponse
+		err = json.UnmarshalRead(resp.Body, &errResp)
+		if err != nil || len(errResp.Errors) == 0 {
+			return nil, "", ctxerr.Errorf(ctx, "unexpected DigiCert status code for POST request: %d", resp.StatusCode)
+		}
+
+		combinedErrorMessages := make([]string, len(errResp.Errors))
+		for i, e := range errResp.Errors {
+			combinedErrorMessages[i] = e.Message
+		}
+		return nil, "", ctxerr.Errorf(ctx, "unexpected DigiCert status code for POST request: %d, errors: %s", resp.StatusCode,
+			strings.Join(combinedErrorMessages, "; "))
+	}
+
+	type certificateResponse struct {
+		SerialNumber   string `json:"serial_number"`
+		DeliveryFormat string `json:"delivery_format"`
+		Certificate    string `json:"certificate"`
+	}
+
+	var certResp certificateResponse
+	err = json.UnmarshalRead(resp.Body, &certResp)
+	if err != nil {
+		return nil, "", ctxerr.Wrap(ctx, err, "unmarshaling DigiCert POST response")
+	}
+
+	if certResp.DeliveryFormat != "x509" {
+		return nil, "", ctxerr.Errorf(ctx, "unexpected DigiCert delivery format: %s", certResp.DeliveryFormat)
+	}
+
+	if len(certResp.Certificate) == 0 {
+		return nil, "", ctxerr.Errorf(ctx, "did not receive DigiCert certificate")
+	}
+
+	level.Debug(logger).Log("msg", "DigiCert certificate created", "serial_number", certResp.SerialNumber)
+
+	// Decode the certificate from PEM format
+	certBlock, _ := pem.Decode([]byte(certResp.Certificate))
+	if certBlock == nil {
+		return nil, "", ctxerr.Errorf(ctx, "failed to decode certificate PEM block")
+	}
+
+	cert, err := x509.ParseCertificate(certBlock.Bytes)
+	if err != nil {
+		return nil, "", ctxerr.Wrap(ctx, err, "parsing certificate from PEM")
+	}
+
+	// Encode the private key and certificate into PKCS12
+	password, err = server.GenerateRandomText(10)
```

Comment:
Double-checking, I know it's often recommended to use at least 16 bytes for e.g. password reset token or email verification token, is 10 bytes enough for a strong password in this scenario? (maybe it is, I honestly don't know, or maybe it's a requirement, in which case fine of course)